src/shims/unix/linux/eventfd.rs

```diff
@@ -217,8 +238,86 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             counter: Cell::new(val.into()),
             is_nonblock,
             clock: RefCell::new(VClock::default()),
+            blocked_read_tid: RefCell::new(None),
+            blocked_write_tid: RefCell::new(None),
         });
 
         interp_ok(Scalar::from_i32(fd_value))
     }
 }
+
+/// Callback function after eventfd read unblocks.
+fn blocking_eventfd_read_callback<'tcx>(
```

Comment:
For future PRs: it is easier to review if you split things out into functions with no changes in a commit first, and then adjust the function to suit your additional use case

Reply:
Good suggestion. I actually ended up opening the original copy and compare it side by side to make sure it is correct because I did quite a lot of change in one go. So this minor refactoring might need a bit more attention.